yarn-project/pxe/src/pxe_oracle_interface/index.ts

```diff
@@ -882,6 +838,15 @@ export class PXEOracleInterface implements ExecutionDataProvider {
   copyCapsule(contractAddress: AztecAddress, srcSlot: Fr, dstSlot: Fr, numEntries: number): Promise<void> {
     return this.capsuleDataProvider.copyCapsule(contractAddress, srcSlot, dstSlot, numEntries);
   }
+
+  async getSharedSecret(address: AztecAddress, ephPk: Point): Promise<Point> {
+    const recipientCompleteAddress = await this.getCompleteAddress(address);
+    const ivskM = await this.keyStore.getMasterSecretKey(
+      recipientCompleteAddress.publicKeys.masterIncomingViewingPublicKey,
+    );
+    const addressSecret = await computeAddressSecret(await recipientCompleteAddress.getPreaddress(), ivskM);
+    return deriveEcdhSharedSecret(addressSecret, ephPk);
```

Comment:
This should likely be a function, and we should have a test in which we check that the secret we get by doing this is the same that we'd get by multipying the ephimeral secret key with the recipient's public key. There's a noir test somewhere that does this iirc, somewhere in the AES utils.

Reply:
Isn't the plan to refactor this realtively soon after this PR to returned the siloed address secret? If not, agree that it makes sense to add the test.